src/routes/books/books.controller.js

```diff
@@ -0,0 +1,108 @@
+const fs = require("fs");
+const { promisify } = require("util");
+const { validationResult } = require("express-validator");
+
+const booksData = require("../../../db/greatreads.data.json");
+
+const writeFile = promisify(fs.writeFile);
+
+const listBooks = (req, res) => {
+  return res.json({
+    data: booksData
+  });
+};
+
+const postBook = async (req, res) => {
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) {
+    return res.status(422).json({ errors: errors.array() });
+  }
+
+  const bookID = (
+    parseInt(booksData.data[booksData.data.length - 1].bookID) + 1
+  ).toString();
```

Comment:
Is it necessary to store ID as string? Integer operations are generally more efficient than string operations; for things that are used to compare between entries, such as ID, it's a good idea to stick to numbers unless there's a good reason not to.